llvm/lib/Target/AIE/AIE2Subtarget.cpp

```diff
@@ -76,3 +76,53 @@ const RegisterBankInfo *AIE2Subtarget::getRegBankInfo() const {
 InstructionSelector *AIE2Subtarget::getInstructionSelector() const {
   return InstSelector.get();
 }
+
+std::bitset<32>
+AIE2Subtarget::getMemoryBanksFromAddressSpace(unsigned AddrSpace) const {
+  using namespace AIE2;
+  std::bitset<32> MemoryBanks;
+
+  switch (static_cast<AddressSpaces>(AddrSpace)) {
+  case AddressSpaces::a:
+    MemoryBanks.set(static_cast<int>(DMBanks::A));
+    break;
+  case AddressSpaces::b:
+    MemoryBanks.set(static_cast<int>(DMBanks::B));
+    break;
+  case AddressSpaces::c:
+    MemoryBanks.set(static_cast<int>(DMBanks::C));
+    break;
+  case AddressSpaces::d:
+    MemoryBanks.set(static_cast<int>(DMBanks::D));
+    break;
+  case AddressSpaces::ab:
+    MemoryBanks.set(static_cast<int>(DMBanks::A))
+        .set(static_cast<int>(DMBanks::B));
+    break;
+  case AddressSpaces::ac:
+    MemoryBanks.set(static_cast<int>(DMBanks::A))
+        .set(static_cast<int>(DMBanks::C));
+    break;
+  case AddressSpaces::ad:
+    MemoryBanks.set(static_cast<int>(DMBanks::A))
+        .set(static_cast<int>(DMBanks::D));
+    break;
+  case AddressSpaces::bc:
+    MemoryBanks.set(static_cast<int>(DMBanks::B))
+        .set(static_cast<int>(DMBanks::C));
+    break;
+  case AddressSpaces::bd:
+    MemoryBanks.set(static_cast<int>(DMBanks::B))
+        .set(static_cast<int>(DMBanks::D));
+    break;
+  case AddressSpaces::cd:
+    MemoryBanks.set(static_cast<int>(DMBanks::C))
+        .set(static_cast<int>(DMBanks::D));
+    break;
+  default:
+    // For unimplemented cases assume none
```

Comment:
I think we need to think about this point! `none` here means empty bitmask and no colision inside the slot. Do we need a pessimistic or optimistic treatment? What do you think @martien-de-jong and @gbossu.

Reply:
I think that if bd means b or d, then none means a or b or c or d, but we implement bd as b and d, blocking both b and d when it occurs. I'm still waiting for a clarifying comment as to what the user expects. Since none means 'no information', my feeling is that it should conflict with everything, so have 0xf as a mask.

Reply:
Assuming equal likelihoods, AB and BC only conflict in one {BB} in four {AB, AC, BB, BC} cases, yet we assume a conflict, and may force an extra cycle.

Reply:
Hi @martien-de-jong ,
Kindly refer to the communication done in https://github.com/Xilinx/llvm-aie/pull/95#discussion_r1663956837